src/nvim/api/autocmd.c

```diff
@@ -281,9 +282,18 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
           PUT(autocmd_info, "desc", CSTR_TO_OBJ(ac->desc));
         }
 
-        PUT(autocmd_info,
-            "command",
-            STRING_OBJ(cstr_as_string(aucmd_exec_to_string(ac, ac->exec))));
+        if (ac->exec.type == CALLABLE_CB) {
+          Callback callback = ac->exec.callable.cb;
+          if (callback.type == kCallbackLua) {
+            if (nlua_ref_is_function(callback.data.luaref)) {
+              PUT(autocmd_info, "callback", LUAREF_OBJ(api_new_luaref(callback.data.luaref)));
+            }
+          }
+        } else {
```

Comment:
I think a `switch` statement would be better here so that we explicitly enumerate all of the cases:

```c
switch (ac->exec.type) {
case CALLABLE_CB:
  // handle callback case
  break;
case CALLABLE_EX:
  PUT(autocmd_info,
      "command",
      STRING_OBJ(cstr_as_string(aucmd_exec_to_string(ac, ac->exec))));
  break;
case CALLABLE_NONE:
  break;
}
```